### PR TITLE
Apply CoreData

### PR DESCRIPTION
**Pull request by Topia99**

### CODE REVIEW

File: Travel-Planner/Travel-Planner/ActivityViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+    }
+    
+    func addActivity(title: String, location: String, time: Date?, notes: String, activityType: ActivityType) {
+        let newActivity = ActivityEntity(context: manager.context)
+        newActivity.title = title
+        newActivity.location = location
+        newActivity.time = time
+        newActivity.notes = notes
+        newActivity.type = activityType
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/ActivityViewModel.swift]: add created at var to activityEntity

File: Travel-Planner/Travel-Planner/ActivityViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        newActivity.dayPlan = self.dayPlan
+        
+        save()
+    }
+    
+    func deleteActivity(_ activity: ActivityEntity) {
+        manager.context.delete(activity)
+        save()
+    }
+
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/ActivityViewModel.swift]: add a update func

File: Travel-Planner/Travel-Planner/CoreData/ActivityEntity+CoreDataProperties.swift [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @NSManaged public var createdAt: Date?
+    @NSManaged public var location: String?
+    @NSManaged public var notes: String?
+    @NSManaged public var time: Date?
+    @NSManaged public var title: String
+    @NSManaged public var type: ActivityType
+    @NSManaged public var dayPlan: DayPlanEntity?
+    @NSManaged public var trip: TripEntity?
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/CoreData/ActivityEntity+CoreDataProperties.swift]: make sure to link activityEntity to TripEntity in ther add func

File: Travel-Planner/Travel-Planner/CoreData/ActivityEntity+CoreDataProperties.swift [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        return NSFetchRequest<ActivityEntity>(entityName: "ActivityEntity")
+    }
+
+    @NSManaged public var createdAt: Date?
+    @NSManaged public var location: String?
+    @NSManaged public var notes: String?
+    @NSManaged public var time: Date?
+    @NSManaged public var title: String
+    @NSManaged public var type: ActivityType
+    @NSManaged public var dayPlan: DayPlanEntity?
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/CoreData/ActivityEntity+CoreDataProperties.swift]: DayPlanEntity should be non_optional

File: Travel-Planner/Travel-Planner/CoreData/DayPlanEntity+CoreDataProperties.swift [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    @nonobjc public class func fetchRequest() -> NSFetchRequest<DayPlanEntity> {
+        return NSFetchRequest<DayPlanEntity>(entityName: "DayPlanEntity")
+    }
+
+    @NSManaged public var date: Date
+    @NSManaged public var dayNumber: Int16
+    @NSManaged public var notes: String?
+    @NSManaged public var title: String?
+    @NSManaged public var activities: NSSet?
+    @NSManaged public var trip: TripEntity?
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/CoreData/DayPlanEntity+CoreDataProperties.swift]: TripEntity Should be non_optional

File: Travel-Planner/Travel-Planner/DayPlanViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+//
+
+import Foundation
+import CoreData
+
+class DayPlanViewModel: ObservableObject {
+    let manager = CoreDataManager.instance
+    let trip: TripEntity
+    @Published var dayPlans: [DayPlanEntity] = []
+    @Published var activities: [ActivityEntity] = []
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/DayPlanViewModel.swift]: remove activities

File: Travel-Planner/Travel-Planner/Models/DayPlan.swift [hunk 7]
```diff
@@ -7,29 +7,29 @@
...
-
-
-struct Item: Identifiable, Codable{
-    var id = UUID()
-    var title: String
-    var location: String?
-    var time: Date?
-    var description: String?
-    
-    var type: ItemType
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Models/DayPlan.swift:33]: remove this file

File: Travel-Planner/Travel-Planner/Models/Trip.swift [hunk 8]
```diff
@@ -6,16 +6,29 @@
...
-    var title: String
-    var startDate: Date
-    var endDate: Date
-    var destinations: [String]
-    var dayPlans: [DayPlan]
-}
+//struct Trip: Identifiable, Codable {
+//    var id = UUID()
+//    var title: String
+//    var startDate: Date
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Models/Trip.swift]: remove file

File: Travel-Planner/Travel-Planner/Models/TripProvider.swift [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+//  TripProvider.swift
+//  Travel-Planner
+//
+//  Created by Jason Zeng on 11/6/24.
+// Help with manage and interact with data in data stack
+
+//import Foundation
+//import CoreData
+//
+//
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Models/TripProvider.swift]: remove file

File: Travel-Planner/Travel-Planner/TripViewModel.swift [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    func getTrips() {
+        let request = NSFetchRequest<TripEntity>(entityName: "TripEntity")
+        
+        do {
+            trips = try manager.context.fetch(request)
+        } catch let error {
+            print("Error fetching trips. \(error.localizedDescription)")
+        }
+    }
+
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/TripViewModel.swift]: add removeTrip and updateTrip func

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanTitleAndDescriptionView.swift [hunk 11]
```diff
@@ -4,27 +4,27 @@
...
-            
-            Spacer()
-        }
-        .font(.title2)
-        .fontWeight(.semibold)
-        
-        
-        TextField("Day Description", text: $dayPlan.description)
-            .font(.body)
-            .padding(.top, 3)
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanTitleAndDescriptionView.swift:27]: DayPlan still need a title and description, find a way to implement it. Add a edit button on the navigationbar trailling to edit trip info?

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift [hunk 12]
```diff
@@ -9,55 +9,54 @@ import SwiftUI
 
 struct DayPlanView: View {
     
-    @Binding var dayPlan: DayPlan
-    @State var isShowingItemDetail: Bool = false // Controls the presentation of the TripItemDetail sheet.
-    @State private var itemToEdit: Item? = nil // Holds the Item to be edited; nil if adding a new item.
+    @StateObject var vm: ActivityViewModel
+    @State var isShowingAddActivity: Bool = false // Controls the presentation of the TripItemDetail sheet.
+//    @State private var itemToEdit: Item? = nil // Holds the Item to be edited; nil if adding a new item.
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift]: clean variables that not use

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift [hunk 13]
```diff
@@ -9,55 +9,54 @@ import SwiftUI
...
             VStack {
                 // Display the date of a dayplan. Ex: Nov 25
-                DayPlanHeadingView(date: dayPlan.date, dayNumber: dayPlan.dayNumber)
+                DayPlanHeadingView(date: vm.dayPlan.date, dayNumber: vm.dayPlan.dayNumber)
                 
                 
-                // TextField input view for Title and description
-                DayPlanTitleAndDescriptionView(dayPlan: $dayPlan)
+//                // TextField input view for Title and description
+//                DayPlanTitleAndDescriptionView(dayPlan: $dayPlan)
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift]: figure it out how to make it work

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift [hunk 14]
```diff
@@ -9,55 +9,54 @@ import SwiftUI
...
                 
                 
                 
+//                // Item List
+//                DayPlanItemListView(items: $dayPlan.items) { item in
+//                    itemToEdit = item // When a user taps on an existing item in the list, itemToEdit is set to the tapped item.
+//                    isShowingItemDetail = true
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift]: clean code

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift [hunk 15]
```diff
@@ -9,55 +9,54 @@ import SwiftUI
...
+                    VStack{
+                        ForEach(vm.activities) { activity in
+                            NavigationLink(destination: ActivityDetailView(activity: activity, vm: vm)) {
+                                ItemCellView(activity: activity)
+                            }
+                        }
+                    }
                 }
                 
                 // Add new item button
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift]: add new activity button

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift [hunk 16]
```diff
@@ -9,55 +9,54 @@ import SwiftUI
...
                 
                 
+//                // Item List
+//                DayPlanItemListView(items: $dayPlan.items) { item in
+//                    itemToEdit = item // When a user taps on an existing item in the list, itemToEdit is set to the tapped item.
+//                    isShowingItemDetail = true
+//                }
+                
+                
                 // Item List
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlan/DayPlanView.swift]: activity list

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/EditActivityView.swift [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+    @ObservedObject var activity: ActivityEntity
+    @StateObject var vm: ActivityViewModel
+    @Environment(\.dismiss) var dismiss
+    
+    
+    @State var titleTextField: String = ""
+    @State var locationTextField: String = ""
+    @State var selectedActivityType: ActivityType = .activity
+    @State var time: Date = Date()
+    @State var notesTextField: String = ""
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/EditActivityView.swift]: try to combine addActivity and editActivity

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/EditActivityView.swift [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+                Button("Cancel") {
+                    dismiss()
+                }
+                .foregroundStyle(.red)
+            }
+            
+            // Add / Done Button
+            ToolbarItem(placement: .navigationBarTrailing) {
+                Button("Done") {
+                    activity.title = titleTextField
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/EditActivityView.swift]: create a updateActivityFunc for update

File: Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/ItemCellView.swift [hunk 19]
```diff
@@ -9,40 +9,40 @@ import SwiftUI
 
 struct ItemCellView: View {
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/ItineraryView/DayPlanItem/ItemCellView.swift]: change name to ActivityCellView

**Review by Topia99 — COMMENTED** [review]
Reviewed by Jason

**Review by Topia99 — COMMENTED** [review]
Good. Reviewed by Jason

**Review by SamYang10086 — APPROVED** [review]

**Review by SamYang10086 — APPROVED** [review]
pass

**Pull request merged**